### PR TITLE
Feature/moj admin

### PR DESCRIPTION
**Pull request by jimnarey**
Hi - before progressing it would be helpful to have some feedback on the layout, especially in relation to the filter (or any other) sidebar. Are we happy with it on the edge, with this screen width, or should I try to make it horizontal at the top. Font-size also has a bearing. Otherwise, getting close I think to the desired result. Some further tidying of files (esp CSS) will be needed once this is complete.

### CODE REVIEW

**Comment by cliffxuan** [comment]
need to resolve the merge conflict.

**Comment by cliffxuan** [comment]
i don't get the govuk theme somehow. is there some uncommitted change?

**Comment by jimnarey** [comment]
Hi - before progressing it would be helpful to have some feedback on the layout, especially in relation to the filter (or any other) sidebar. Are we happy with it on the edge, with this screen width, or should I try to make it horizontal at the top. Font-size also has a bearing. Otherwise, getting close I think to the desired result. Some further tidying of files (esp CSS) will be needed once this is complete.

**Comment by cliffxuan** [comment]
yes, it was an issue with my local settings. now i can see the theme. looks good!

**Comment by cliffxuan** [comment]
had a play. some thoughts:
1. it's really good overall!
2. on the left of the page, the list has the item `Client`. In stock django admin, the verbose name `Service area` is used instead. We should use verbose name if available.
3. heading of the table is not obvious as it's got the same style as table rows
4. when sorting order is applied, heading of the table should indicate using triangles either pointing up or down. this is in the stock django admin interface.
5. positioning of the `Add` button feels a bit too much to the left, but that could just be me.

File: dashboard/apps/moj_admin/static/moj_admin/js/dwp_admin.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      // Check the href of each link in the sidebar
+      $('.admin-sidebar-content a').each(function () {
+        if (url === $(this).attr('href')) {
+          // If the href matches the current URL set it as active
+          $(this).parents('li').addClass('active');
+        }
+      });
+    });
+    //
+    // // Tab Panes
```
Comment (s-block) [inline dashboard/apps/moj_admin/static/moj_admin/js/dwp_admin.js:18]: Can you remove commented code if it is not needed

File: dashboard/apps/prototype/static/prototype/dist/styles.css [hunk 2]
```diff
@@ -0,0 +1,2380 @@
...ius:0}.select2-container--classic.select2-container--open.select2-container--below .select2-selection--multiple{border-bottom:none;border-bottom-left-radius:0;border-bottom-right-radius:0}.select2-container--classic .select2-search--dropdown .select2-search__field{border:1px solid #aaa;outline:0}.select2-container--classic .select2-search--inline .select2-search__field{outline:0;box-shadow:none}.select2-container--classic .select2-dropdown{background-color:#fff;border:1px solid transparent}.select2-container--classic .select2-dropdown--above{border-bottom:none}.select2-container--classic .select2-dropdown--below{border-top:none}.select2-container--classic .select2-results>.select2-results__options{max-height:200px;overflow-y:auto}.select2-container--classic .select2-results__option[role=group]{padding:0}.select2-container--classic .select2-results__option[aria-disabled=true]{color:grey}.select2-container--classic .select2-results__option--highlighted[aria-selected]{background-color:#3875d7;color:#fff}.select2-container--classic .select2-results__group{cursor:default;display:block;padding:6px}.select2-container--classic.select2-container--open .select2-dropdown{border-color:#5897fb}
```
Comment (cliffxuan) [inline dashboard/apps/prototype/static/prototype/dist/styles.css]: the directory `dashboard/apps/prototype/static/prototype` is no needed and does not exist in branch `develop`. please remove it from this branch. this is the directory layout on `develop`.

```
dashboard/apps/prototype/static
├── dist
│   ├── prototype.css
│   ├── prototype.js
├── js
│   ├── __tests__
│   │   └── project-test.js
│   ├── main.js
│   ├── plotly-custom.js
│   ├── project.js
│   └── service.js
└── styles
    ├── gov-uk-elements.css
    ├── main.css
    └── moj_template_overrides.css
```

File: dashboard/settings/base.py [hunk 3]
```diff
@@ -151,6 +151,12 @@
 
 FLOAT_API_TOKEN = os.environ.get('FLOAT_API_TOKEN')
 
+ADMIN_HEADINGS = {
```
Comment (s-block) [inline dashboard/settings/base.py]: This doesn't look to be used. I think it is left over from my initial start on this

File: dashboard/settings/base.py [hunk 4]
```diff
@@ -37,15 +38,14 @@
...
 
 INSTALLED_APPS = [
+    'moj_admin',
     'django.contrib.admin',
     'django.contrib.auth',
     'django.contrib.contenttypes',
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-
```
Comment (s-block) [inline dashboard/settings/base.py]: no need to remove these lines, they separate django, third party and local.

File: dashboard/settings/base.py [hunk 5]
```diff
@@ -9,6 +9,7 @@
 For the full list of settings and their values, see
 https://docs.djangoproject.com/en/1.9/ref/settings/
 """
+from django.utils.translation import ugettext_lazy
```
Comment (s-block) [inline dashboard/settings/base.py]: this shouldn't be needed if the ADMIN_HEADINGS setting is removed

File: dashboard/apps/moj_admin/templates/moj_admin/base.html [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    {% block extrastyle %}
+
+    {% endblock %}
+{% endblock %}
+
+{% block head %}
+    {% block blockbots %}
+        <meta name="robots" content="NONE,NOARCHIVE"/>
+    {% endblock %}
+    <script type="text/javascript" src="https://code.jquery.com/jquery-2.2.4.min.js"></script>
```
Comment (s-block) [inline dashboard/apps/moj_admin/templates/moj_admin/base.html:28]: moj_base includes a different version of jquery. do we need both?

Comment (cliffxuan) [inline dashboard/apps/moj_admin/templates/moj_admin/base.html:28]: yes, we should just need one. the one on `moj_base` is very old. we need to do something with it.

**Comment by s-block** [comment]
Also can you do an interactive rebase before merging to squash/fixup some of the commits. There are a couple of my initial stash commits in there

File: dashboard/apps/moj_admin/static/moj_admin/js/dwp_admin.js (hunk 1 above)
Comment (jimnarey) [inline dashboard/apps/moj_admin/static/moj_admin/js/dwp_admin.js:18]: It isn't now, but would be to use some of the styling on the DWP admin. Suggest leaving in until Maria has had a look once I've made some of the other changes.

File: dashboard/apps/moj_admin/templates/moj_admin/base.html (hunk 6 above)
Comment (jimnarey) [inline dashboard/apps/moj_admin/templates/moj_admin/base.html:28]: It's the same as in site-packages/moj_template/templates/moj_template/base.html. I had to modify this template and store it as a new file, as it shared some tags with the base django admin template ('content' for example, possibly others). Ideally (I think?) we'd propose the jquery issue and other changes to the moj-template package but happy to deal with as you both think best. Let me know.

Comment (cliffxuan) [inline dashboard/apps/moj_admin/templates/moj_admin/base.html:28]: i see. so you are not deriving from `base.html`. in that case, it's fine. 

i also found `moj_template` out of date and not work well with the govuk css files.

**Pull request merged**